### PR TITLE
[DPM] Map vpcId to hostIp in MulticastGoalStateV2

### PR DESCRIPTION
**Pull request by VanderChen**
In this PR, we map vpcId to hostIp in MulticastGoalStateV2.
1. List<String> replaced by Map<>  for VpcIds in MulticastGoalStateV2.
2. FunctionUnicastGoalStateV2 and FunctionMulticastGoalStateV2 are added for group-node Pulsar implementation, which uses Pulsar Function sending goal state.

### CODE REVIEW

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/MulticastGoalStateV2.java [hunk 1]
```diff
@@ -16,19 +16,16 @@
...
 import com.futurewei.alcor.web.entity.dataplane.MulticastGoalStateByte;
 
-import java.util.ArrayList;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
 
 public class MulticastGoalStateV2 {
     private Set<String> hostIps;
```
Comment (VanderChen) [inline services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/MulticastGoalStateV2.java:27]: In my opinion, Set of hostIps can be removed. The getter of hostIps can be implemented by `hostVpcMap.keySet()`

**Comment by xieus** [comment]
@VanderChen  This PR has conflict with master. 

@DavidLiu506 could you check if you can help resolve the conflict? Thanks.

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/MulticastGoalStateV2.java (hunk 1 above)
Comment (cj-chung) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/MulticastGoalStateV2.java:27]: @VanderChen Let's make a comment here and we can make change in the next release.

Comment (VanderChen) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/MulticastGoalStateV2.java:27]: In MulticastGoalStateV2, a Set holds the hostIps needed by gRPC path and a HashMap holds the (HostIp, VpcId) pairs for Pulsar path. In my opinion, the Set of hostIps can be removed, and HashMap.keySet() can be used to get hostIps for gRPC path.

**Comment by DavidLiu506** [comment]
Works on Jenkins

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/MulticastGoalStateV2.java (hunk 1 above)
Comment (cj-chung) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/MulticastGoalStateV2.java:27]: @VanderChen  I see. If we remove this line of code now, you probably also need to modify lots of code in many different places. If this is the case, we better make a comment here and change it later in another PR.

**Comment by cj-chung** [comment]
Pass testcases in gPRC mode in Jenkins.
For the pulsar mode, we still need to verify it in the Jenkins with MQ environment in the future.

**Review by cj-chung — APPROVED** [review]
LGTM with minor comment. I think we can merge this PR now.

File: services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/MulticastGoalStateV2.java [hunk 2]
```diff
@@ -16,37 +16,41 @@
...
-import java.util.List;
-import java.util.Set;
+import java.util.*;
 
 public class MulticastGoalStateV2{
     private Set<String> hostIps;
-    private List<String> vpcIds;
-    private List<String> topics;
-    private List<String> subTopics;
+    private Map<String, Set<String>> hostVpcMap;
```
[the remark this answers is not in the document]
Comment (VanderChen) [reply services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/entity/MulticastGoalStateV2.java:28]: Comments will be added in the next PR.

**Pull request merged**